crates/ra_ide/src/display/structure.rs

```diff
@@ -151,11 +152,12 @@ fn structure_node(node: &SyntaxNode) -> Option<StructureNode> {
                 Some(node)
             },
             ast::MacroCall(it) => {
-                let first_token = it.syntax().first_token().unwrap();
-                if first_token.text().as_str() != "macro_rules" {
-                    return None;
+                match it.path().and_then(|p| Path::from_ast(p)) {
```

Comment:
We can do this check using only Ast: `it.path().and_then(|it| it.segment()).and_then(|it| it.name_ref()).and_then(|it| it.text() == "macro_rules")`. `AsName` is intended to be a thing mostly private to `hir`.